homeassistant/components/airpatrol/climate.py

```diff
@@ -0,0 +1,321 @@
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up AirPatrol climate entities."""
+    coordinator: AirPatrolDataUpdateCoordinator = config_entry.runtime_data
+    # Create climate entities for each unit
+    entities = []
+    client = await coordinator.async_get_api_client()
+    units = await client.get_data()
+    entities = [
+        AirPatrolClimate(coordinator, unit, unit["unit_id"])
+        for unit in units
+        if "climate" in unit
+    ]
+
+    async_add_entities(entities)
+
+
+class AirPatrolClimate(
+    CoordinatorEntity[AirPatrolDataUpdateCoordinator], ClimateEntity
+):
+    """AirPatrol climate entity."""
+
+    _attr_has_entity_name = True
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.FAN_MODE
+        | ClimateEntityFeature.SWING_MODE
+        | ClimateEntityFeature.TURN_OFF
+        | ClimateEntityFeature.TURN_ON
+    )
+
+    def __init__(
+        self,
+        coordinator: AirPatrolDataUpdateCoordinator,
+        unit: dict[str, Any],
+        unit_id: str,
+    ) -> None:
+        """Initialize the climate entity."""
+        super().__init__(coordinator)
+        self._unit = unit
+        self._unit_id = unit_id
+        unique_id = getattr(coordinator.config_entry, "unique_id", None)
+        self._attr_unique_id = (
+            f"{unique_id}_{unit_id}_climate"
+            if unique_id is not None
+            else f"{unit_id}_climate"
+        )
+        self._attr_translation_key = "climate"
+        self._unavailable_logged = False
+        # Set device info
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, unit_id)},
+            name=unit["name"],
+            manufacturer=unit["manufacturer"],
+            model=unit["model"],
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        is_available = (
+            super().available
+            and self.coordinator.data is not None
+            and any(
+                isinstance(u, dict) and u.get("unit_id") == self._unit_id
+                for u in self.coordinator.data
+            )
+            and "climate" in self._unit
+        )
+        if not is_available and not self._unavailable_logged:
+            _LOGGER.info("The climate entity '%s' is unavailable", self._attr_unique_id)
+            self._unavailable_logged = True
+        elif self._unavailable_logged:
+            _LOGGER.info("The climate entity '%s' is back online", self._attr_unique_id)
+            self._unavailable_logged = False
+        return is_available
+
+    @property
+    def current_humidity(self) -> float | None:
+        """Return the current humidity."""
+        climate_data = self._unit.get("climate", {})
+        if humidity := climate_data.get("RoomHumidity"):
+            try:
+                return float(humidity)
```

Comment:
Can this be something else than a number (if it's not `None`)?
Same question for temp below.

Reply:
Not sure, I haven't seen any other value than a number!

Reply:
Then I guess we can remove this guarding of anything else than a number?

Reply:
What do you mean? Should I remove the try catch?

Reply:
Since you're saying it's always a number if not `None` I think we can remove it.

Reply:
Had to look over my notes and seems as the api returns the values as strings.
```
  "RoomTemp": "27.559",
  "RoomHumidity": "57"
```

Reply:
Yes, so we should use float to convert them but we don't need the try-block.